### ISSUE TITLE
 java.lang.NullPointerException: Attempt to invoke virtual method 'int android.support.v7.widget.OrientationHelper.getMode()' on a null object reference

### ISSUE THREAD

**Issue by Kolyall** [leochuan/ViewPagerLayoutManager#27]
```
   java.lang.NullPointerException: Attempt to invoke virtual method 'int android.support.v7.widget.OrientationHelper.getMode()' on a null object reference
                                                                    at android.support.v7.widget.LinearLayoutManager.resolveIsInfinite(LinearLayoutManager.java:1197)
                                                                    at android.support.v7.widget.LinearLayoutManager.updateLayoutState(LinearLayoutManager.java:1162)
                                                                    at android.support.v7.widget.LinearLayoutManager.collectAdjacentPrefetchPositions(LinearLayoutManager.java:1311)
                                                                    at android.support.v7.widget.GapWorker$LayoutPrefetchRegistryImpl.collectPrefetchPositionsFromView(GapWorker.java:94)
                                                                    at android.support.v7.widget.GapWorker.buildTaskList(GapWorker.java:219)
                                                                    at android.support.v7.widget.GapWorker.prefetch(GapWorker.java:364)
                                                                    at android.support.v7.widget.GapWorker.run(GapWorker.java:396)
                                                                    at android.os.Handler.handleCallback(Handler.java:739)
                                                                    at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                    at android.os.Looper.loop(Looper.java:158)
                                                                    at android.app.ActivityThread.main(ActivityThread.java:7224)
                                                                    at java.lang.reflect.Method.invoke(Native Method)
                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```
Why the error may happen?

**Comment by leochuan** [comment]
duplicate #25 , This error confuse me either, It seems like it has something to do with the version of `RecyclerView`, You may change the version  into `26.0.1`.  Because I can not reappear this error, So I still not quite sure why it happen.

**Comment by Kolyall** [comment]
I found that the error doesnt appear if I also add this:
``` 
GravitySnapHelper snapHelper = new GravitySnapHelper(Gravity.START);
snapHelper.attachToRecyclerView(mRecyclerView);
```
from this lib
https://github.com/rubensousa/RecyclerViewSnap

**Comment by Kolyall** [comment]
But if I set this, the error appears again
```
LinearSnapHelper snapHelper = new LinearSnapHelper();
snapHelper.attachToRecyclerView(mRecyclerView);
```
WTF?

**Comment by leochuan** [comment]
what is your gradle build tools version?

**Comment by Kolyall** [comment]
```
 androidBuildToolsVersion = "26.0.2"
 androidMinSdkVersion = 19
 androidTargetSdkVersion = 23
 androidCompileSdkVersion = 25
 supportLibVer = '25.4.0'
compileOnly "com.android.support:recyclerview-v7:${supportLibVer }"
```
My customer doest want to update this versions

**Comment by leochuan** [comment]
Finally reappear this error..I think I can fix it today

**Comment by leochuan** [comment]
Fixed by : 8c4acfafd9ba663ae900796911df21146ad35e53. Try version 2.0.3.
It because `LinearLayoutManager` below 26.0.0 did not call `ensureLayoutState` before `updateLayoutState` in method `collectAdjacentPrefetchPositions`. Cause I don't need to prefetch items, so I just simply disable it. 

I find that `GravitySnapHelper` use `api "com.android.support:recyclerview-v7:27.0.2"` in its gradle build file. So maybe when you set it, `recyclerview-27.0.2` is the one you use.

**Comment by Kolyall** [comment]
Fixed!
